cognitedata/cdf-fabric-replicator
# Added Time Series to Extractor

Added Functionality for Extracting Time Series Data from Lakehouse into CDF by adding logic and methods into the extractor class. These changes should highlight the following:

- Time Series Extraction from Lakehouse to CDF in Extractor Method
- Logic to handle checkpointing in the state store by time-series timestamp
- Refactored code to help split logic between Extracting Events and Extracting Time-Series  

## How to Verify 

-  in `.Env` file replace the following:
LAKEHOUSE_ABFSS_PREFIX=abfss://FabricCogniteDemo@onelake.dfs.fabric.microsoft.com/CogniteLakehouse.Lakehouse
TEST_CONFIG_PATH=tests/integration/test_config.yaml
EXTRACTOR_EVENT_PATH=Tables/TestEvents
EXTRACTOR_FILE_PATH=Files/tank_thumbnail.png
EXTRAXTOR_RAW_TS_PATH=Tables/mtu_historic
EXTRACTOR_DATASET_ID=3351080165451477
EXTRACTOR_TS_PREFIX=ktst_new_extract_
EXTRACTOR_DESTINATION_TYPE=events
-  Run the project with the command in the terminal `poetry run cdf_fabric_replicator example_config.yaml `
- After running you should double check the CDF portal to see the time-series data points being populated:

![image](https://github.com/kenny-chalupa/cdf-fabric-replicator/assets/50754008/aba99319-b5fe-4a29-8c5f-3e52435b94b1)
